### PR TITLE
Allow custom styled input size

### LINKED ISSUE #5402: StyledTextInput doesn't work if props size is a custom value
### Expected Behavior

Any component that uses StyledTextInput should accept custom size values.
E.g:
```ts
<Grommet>
    <DateInput size="1rem" />
    <MaskedInput size="1em" />
    <TextInput size="16px" />
    <TextArea size="60%" />
</Grommet>
```

### Actual Behavior
The component is not rendered and the "TypeError: data is undefined" error is raised.

### URL, screen shot, or Codepen exhibiting the issue
https://codesandbox.io/s/grommet-v2-template-forked-lq58q?file=/index.js

### Steps to Reproduce

1. Import one of these components: FileInput, MaskedInput, TextInput, TextArea
2. Set the size props to any custom value as: "16px", "1rem" and so on.
3. The component will not render

### Note

As I've already worked on #5382 issue,  I'd to work on this issue as well.

Edit: Replaced FileInput by DateInput at it seems that the size props doesn't changes the font size of FileInput

### PR DESCRIPTION
**Pull request by LucasCarrias**
#### What does this PR do?
Allow inputs to have custom inputs sizes.
```ts
<DateInput size="1rem" />
<MaskedInput size="1em" />
<TextInput size="16px" />
<TextArea size="60%" />
```
#### Where should the reviewer start?
Only the changes at **styles.js**
#### What testing has been done on this PR?
Tests if **TextInput, TextArea, DateInput, MaskedInput** renders with the default and custom sizes values
#### How should this be manually tested?
Tests if the inputs sizes renders and changes if  
#### What are the relevant issues?
Closes #5402

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
DateInput should have the `size` prop added to the documentation. TextInput and TextArea should have the `size` prop in the documentation described as "The size of the text"

**Comment by LucasCarrias** [comment]
@jcfilben Thanks for the review. I'll update that accordingly.

**Review by jcfilben — APPROVED** [review]

**Review by jcfilben — COMMENTED** [review]
Changes look good, just fix merge conflicts

File: src/js/components/DateInput/doc.js [hunk 1]
```diff
@@ -59,6 +59,10 @@ export const doc = DateInput => {
       PropTypes.string,
       PropTypes.arrayOf(PropTypes.string),
     ]).description('The date or date range value(s) in ISO8601 format.'),
+    size: PropTypes.oneOfType([
+      PropTypes.oneOf(['small', 'medium', 'large', 'xlarge']),
```
[the remark this answers is not in the document]
Comment (LucasCarrias) [reply src/js/components/DateInput/doc.js:63]: I've just followed the MaskedInput, TextArea and TextInput docs patterns. I wasn't quite sure if all sizes were required to be updated in the documentation.

As they seem to be required, I'll add all remaining text sizes into those components docs.

**Review by jcfilben — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**